### DISCUSSION TITLE
Support multiple platforms in Homebrew formula

### DISCUSSION THREAD

**Discussion by int128** [renovatebot/renovate#22204] [Suggest an Idea]
**What would you like Renovate to be able to do?**

I'd like to update the versions of multiple platforms in a formula. Currently `homebrew` manager updates only first match, for example, https://github.com/int128/homebrew-tap/pull/2. Also it would be nice to update `version` line.

Here is an example of expected diff:

```diff
 class Example < Formula
   desc "example"
-  version "v0.1.0"
+  version "v0.2.0"
   on_macos do
-    url "https://github.com/int128/kubectl-socat/releases/download/v0.1.0/kubectl-socat_v0.1.0_darwin_amd64.zip"
-    sha256 "295b650da9766fc4e7ccc80201223f2366b800c419cc3bfab804a2d6e87f4cc0"
+    url "https://github.com/int128/kubectl-socat/releases/download/v0.2.0/kubectl-socat_v0.2.0_darwin_amd64.zip"
+    sha256 "2809248e54d655cf13ddc896df51cb0b1e187a9193f2e7a310f75299b3821250"
   end
   on_linux do
-    url "https://github.com/int128/kubectl-socat/releases/download/v0.1.0/kubectl-socat_v0.1.0_linux_amd64.zip"
-    sha256 "a0862211648202ba80e3c33078ab1c8c2821dc6419e5fa8c496e83500cbfcd34"
+    url "https://github.com/int128/kubectl-socat/releases/download/v0.2.0/kubectl-socat_v0.2.0_linux_amd64.zip"
+    sha256 "f121fbfcbc01d893767d7fd31c7f4b62bb1db729ba0cd8eab2f46abec90ffa8e"
   end
```

**Did you already have any implementation ideas?**

I have investigated the current implementation and it depends on the following URL formats:

- `https://github.com/ONWER/NAME/archive/VERSION.tar.gz`
- `https://github.com/OWNER/NAME/releases/download/VERSION/NAME-VERSION.tar.gz`

https://github.com/renovatebot/renovate/blob/7fe97bfd2c9c811c2723cb441226a1d41fbd11b6/lib/manager/homebrew/update.ts#L162-L164

It need to be changed like:

- `https://github.com/ONWER/NAME/archive/VERSION.tar.gz`
- `https://github.com/OWNER/NAME/releases/download/VERSION/` + replaceAll(oldVersion, newVersion)

Also it need to update multiple URLs but I'm not sure how to do it.

I'm happy to contribute this change. Thank you.

**Comment by viceice** [comment]
I think we first need to refactor the manager to use artifact handling and update the urls and sha256 in updateArtifact function.

**Comment by jackton1** [comment]
What’s the current status of this? Curious it’s been open for a while now
